hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/MergeIntoHoodieTableCommand.scala

```diff
@@ -659,7 +659,7 @@ case class MergeIntoHoodieTableCommand(mergeInto: MergeIntoTable) extends Hoodie
       CANONICALIZE_SCHEMA.key -> "false",
       SCHEMA_ALLOW_AUTO_EVOLUTION_COLUMN_DROP.key -> "true",
       HoodieSparkSqlWriter.SQL_MERGE_INTO_WRITES.key -> "true",
-      HoodieWriteConfig.SPARK_SQL_MERGE_INTO_PREPPED_KEY -> enableOptimizedMerge,
+      HoodieWriteConfig.SPARK_SQL_MERGE_INTO_PREPPED_KEY -> isPrimaryKeylessTable.toString,
```

Comment:
So, we are not dis-allowing `optimizedMerge` ? 
i.e. prior to this patch, users can explicitly set value for SPARK_SQL_OPTIMIZED_WRITES config to true and leverage prepped flow benefits. 

so, is it a conscious decision to deprecate it?

Reply:
Outside auto keygen feature the "optimized merge" causes regression so it is intentionally disabled.